Content.Client/UserInterface/Systems/CloseWindow/CloseRecentWindowUIController.cs

```diff
@@ -0,0 +1,124 @@
+using Content.Client.Gameplay;
+using Content.Client.Info;
+using Robust.Client.Input;
+using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.Controllers;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Shared.Input;
+using Robust.Shared.Input.Binding;
+
+namespace Content.Client.UserInterface.Systems.Info;
+
+public sealed class CloseRecentWindowUIController : UIController
+{
+    [Dependency] private readonly IInputManager _inputManager = default!;
+    [Dependency] private readonly IUserInterfaceManager _uiManager = default!;
+
+    /// <summary>
+    /// A list of windows that have been interacted with recently.  Windows should only
+    /// be in this list once, with the most recent window at the end, and the oldest
+    /// window at the start.
+    /// </summary>
+    List<BaseWindow> recentlyInteractedWindows = new List<BaseWindow>();
+
+    public override void Initialize()
+    {
+        // Add listeners to be able to know when windows are opened.
+        // Note -- I am assuming this does not need to be unlistened since UIControllers live
+        // forever(?), but if there's a correct way to unlisten, please let me know :)
+        _uiManager.OnKeyBindDown += OnKeyBindDown;
+        _uiManager.WindowRoot.OnChildAdded += OnRootChildAdded;
+
+        _inputManager.SetInputCommand(EngineKeyFunctions.WindowCloseRecent,
+            InputCmdHandler.FromDelegate(session => CloseMostRecentWindow()));
+    }
+
+    private void CloseMostRecentWindow()
+    {
+        // Search backwards through the recency list to find a still open window and close it
+        for (int i=recentlyInteractedWindows.Count-1; i>=0; i--)
+        {
+            BaseWindow window = recentlyInteractedWindows[i];
```

Comment:
We prefer implicit types (var) in this codebase

```suggestion
            var window = recentlyInteractedWindows[i];
```